### PR TITLE
implement pathfinding

### PR DESCRIPTION
**Pull request by ThierryBerger**

### CODE REVIEW

File: src/path.rs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    Some(res)
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::grid::Grid;
+
+    #[test]
+    fn grid_pathfinding() {
```
Comment (ThierryBerger) [inline src/path.rs:33]: this is not really testing anything useful, but it was useful as a playground to understand the library pathfinding, as well as grid from this project

File: src/path.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    let mut res = vec![];
+    for i in 1..goals.len() {
+        let prev_goal = goals[i - 1];
+        let goal = goals[i];
+        let to_add = bfs(
+            &prev_goal,
+            |p| {
+                let mut succ = grid.neighbours(*p);
+                succ.retain(|x| match x {
+                    // FIXME: map borders exist to avoid infinite search.
```
Comment (ThierryBerger) [inline src/path.rs:14]: See https://discord.com/channels/691052431525675048/891721828123414538/891808675512090635

File: src/path.rs [hunk 3]
```diff
@@ -0,0 +1,63 @@
+use pathfinding::prelude::bfs;
+
+use crate::grid::Grid;
+
+#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct Pos(i32, i32);
```
Comment (ThierryBerger) [inline src/path.rs]: that struct is actually useless

File: src/path.rs [hunk 4]
```diff
@@ -0,0 +1,44 @@
+use crate::grid::Grid;
+use pathfinding::prelude::bfs;
+
+pub fn resolve(grid: &Grid, goals: &[(i32, i32)]) -> Option<Vec<(i32, i32)>> {
+    let mut res = vec![];
+    for i in 1..goals.len() {
+        let prev_goal = goals[i - 1];
+        let goal = goals[i];
+        let to_add = bfs(
```
Comment (Sheepyhead) [inline src/path.rs:9]: Why use BFS instead of A*?

File: src/path.rs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            &prev_goal,
+            |p| {
```
Comment (Sheepyhead) [inline src/path.rs:11]: Please avoid these single letter names unless it's glaringly obvious what the binding is

File: src/path.rs [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                let mut succ = grid.neighbours(*p);
+                succ.retain(|x| match x {
```
Comment (Sheepyhead) [inline src/path.rs:13]: See previous comment, use a more descriptive name

File: src/path.rs [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+                    // FIXME: map borders exist to avoid infinite search.
+                    (-100..=100, -100..=100) => grid.get(*x).is_none(),
+                    _ => false,
+                });
+                succ
+            },
+            |p| *p == goal,
```
Comment (Sheepyhead) [inline src/path.rs:20]: Naming

File: src/path.rs (hunk 1 above)
Comment (Sheepyhead) [reply src/path.rs:33]: I think it should be removed then, as temporary debug code.

**Review by Sheepyhead — CHANGES_REQUESTED** [review]
Got a few styling concerns, nothing semantic though, nice work!

File: src/path.rs (hunk 2 above)
Comment (Sheepyhead) [reply src/path.rs:14]: Yeah I'll settle on map bounds eventually, this is a perfectly fine temporary measure

**Comment by Sheepyhead** [comment]
As an addendum clippy pings two minor semantic issues that should be resolved, I believe it suggests fixes for both. Idk what editor you use but it should work with the .vscode settings in the project unless you don't have clippy installed

**Pull request merged**